libs/vm/include/vm/state.hpp

```diff
@@ -285,28 +291,40 @@ inline Ptr<IState> IState::Construct(VM *vm, TypeId type_id, Args &&... args)
   }  // switch
 }
 
-inline Ptr<IState> IState::Constructor(VM *vm, TypeId type_id, Ptr<String> const &name,
+template <typename... Args>
+inline Ptr<IState> IState::Construct(VM *vm, TypeId type_id, Args &&... args)
+{
+  TypeInfo const &type_info     = vm->GetTypeInfo(type_id);
+  TypeId const    value_type_id = type_info.parameter_type_ids[0];
+  return ConstructIntrinsic(vm, type_id, value_type_id, std::forward<Args>(args)...);
+}
+
+inline Ptr<IState> IState::Constructor(VM *vm, TypeId type_id, Ptr<Object> &&name,
                                        TemplateParameter const &value)
 {
   if (name)
   {
-    return Construct(vm, type_id, name, value);
+    return Construct(vm, type_id, std::move(name), value);
```

Comment:
std::forward

Reply:
The `std::forward<T>(...)` makes sense only in perfect forwarding case, what requires the method being templated with `T` type as method template parameter.
In this case the method is not templated (we are dealing here with specific (upfront known) type `Ptr<Object>`).